src/shared/index.js

```diff
@@ -32,10 +32,15 @@ async function generate(date, options = {}) {
   }
 
   // Crawler
-  const output = scrapeData(srcs)
-    .then(writeRawRegression)
-    // processor
-    .then(rateSources)
+  const output = await scrapeData(srcs).then(writeRawRegression);
+
+  if (options.onlyScrape) {
+    console.log('Exiting after scrape');
+    return output;
+  }
+
+  // processor
+  rateSources(output)
```

Comment:
You needed to `await` this, otherwise this function returns before the promise resolves.